hq_superset/hq_domain.py

```diff
@@ -64,6 +77,28 @@ def ensure_domain_selected():
         return redirect(url_for('SelectDomainView.list', next=request.url))
 
 
+def sync_user_domain_role():
+    if is_user_admin() or (
+        request.url_rule
+        and request.url_rule.endpoint in DOMAIN_EXCLUDED_VIEWS
+    ):
+        return
+    if _domain_role_expired():
+        _sync_domain_role()
+
+
+def _domain_role_expired():
+    if not session.get(SESSION_DOMAIN_ROLE_LAST_SYNCED_AT):
+        return True
+
+    time_since_last_sync = datetime_utcnow_naive() - session[SESSION_DOMAIN_ROLE_LAST_SYNCED_AT]
```

Comment:
@mkangia 
Nit: for more robust implementation we could check if `session[SESSION_DOMAIN_ROLE_LAST_SYNCED_AT]` is a datetime-like object that could be operated on. 

How about:
```
last_sync_datetime = session[SESSION_DOMAIN_ROLE_LAST_SYNCED_AT]
if not isinstance(last_sync_datetime, datetime):
    return False
```

Reply:
Cool. I can add that. This value is only always set by CCA and can't edited by the user so it would certainly be a datetime but no harm in adding safety.